### PR TITLE
Move management of LevelDB and MapDB inside the KV module

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: kv/src/main/kotlin/net/consensys/cava/kv/LevelDBKeyValueStore.kt [hunk 1]
```diff
@@ -18,15 +18,20 @@ package net.consensys.cava.kv
...
+import java.nio.file.Path
 
 /**
  * A key-value store backed by LevelDB.
  */
-class LevelDBKeyValueStore(private val levelDB: DB) : KeyValueStore {
+class LevelDBKeyValueStore(databasePath: Path) : KeyValueStore {
+
+  private val db = JniDBFactory.factory.open(databasePath.toFile(),
+    Options().createIfMissing(true).cacheSize((100 * 1048576).toLong()))
```
Comment (cleishm) [inline kv/src/main/kotlin/net/consensys/cava/kv/LevelDBKeyValueStore.kt]: hardcoded cache size? Should probably be passed in (with a default). Are there other options one might want to set? If so, a builder might be appropriate...

File: kv/src/main/kotlin/net/consensys/cava/kv/LevelDBKeyValueStore.kt [hunk 2]
```diff
@@ -35,6 +40,11 @@ class LevelDBKeyValueStore(private val levelDB: DB) : KeyValueStore {
...
 
   override suspend fun put(key: Bytes, value: Bytes) = withContext(Unconfined) {
-    levelDB.put(key.toArrayUnsafe(), value.toArrayUnsafe())
+    db.put(key.toArrayUnsafe(), value.toArrayUnsafe())
   }
+
+  /**
+   * Closes the underlying LevelDB instance.
+   */
+  fun close() = db.close()
```
Comment (cleishm) [inline kv/src/main/kotlin/net/consensys/cava/kv/LevelDBKeyValueStore.kt]: Should the class implement Closable/AutoCloseable?

File: kv/src/main/kotlin/net/consensys/cava/kv/MapDBKeyValueStore.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import org.mapdb.DataOutput2
+import java.nio.file.Path
+
+/**
+ * A key-value store backed by a MapDB instance.
+ */
+class MapDBKeyValueStore(databasePath: Path) : KeyValueStore {
+
+  private val db = DBMaker.fileDB(databasePath.toFile()).transactionEnable().closeOnJvmShutdown().make()
+  private val storageData = db.hashMap("storageData", BytesSerializer(), BytesSerializer()).createOrOpen()
```
Comment (cleishm) [inline kv/src/main/kotlin/net/consensys/cava/kv/MapDBKeyValueStore.kt]: What's the "storageData" argument, and should that be configurable?

File: kv/src/main/kotlin/net/consensys/cava/kv/LevelDBKeyValueStore.kt (hunk 1 above)
Comment (atoulme) [reply kv/src/main/kotlin/net/consensys/cava/kv/LevelDBKeyValueStore.kt]: Yes, there are other options.
Options itself uses the builder approach. I can change the constructor to take an Options object.

**Pull request merged**